### PR TITLE
feat: add config option to highlight dir with devicons active

### PR DESCRIPTION
**Pull request by opalmay**
only makes a difference with devicons on.
highlight_dirnames_with_devicons = false:
image
highlight_dirnames_with_devicons = true:
image

### CODE REVIEW

**Comment by tamago324** [comment]
@opalmay 
Thank you very much.
I would like the configuration to be structured like this:

```lua
require'lir'.setup {
  devicons = {
    enable = false,
    highlight_dirnames = false
  },
  ...
```

If it seems difficult, I would like to implement it myself.

**Comment by opalmay** [comment]
> @opalmay Thank you very much. I would like the configuration to be structured like this:
> 
> ```lua
> require'lir'.setup {
>   devicons = {
>     enable = false,
>     highlight_dirnames = false
>   },
>   ...
> ```
> 
> If it seems difficult, I would like to implement it myself.

Done :)
Beware that it's a breaking change.

**Comment by tamago324** [comment]
Please add processing when the `devicons` setting value is empty and a warning message when the `devicons_enable` setting value is specified.

```diff
  function config.set_default_values(opts)
    config.values = vim.tbl_deep_extend("force", defaults_values, opts or {})
  
    -- For an empty table
    if vim.tbl_isempty(config.values.float) then
      config.values.float = defaults_values.float
    elseif vim.tbl_isempty(config.values.float.curdir_window) then
      config.values.float.curdir_window = defaults_values.float.curdir_window
    end
  
+   if vim.tbl_isempty(config.values.devicons) then
+     config.values.devicons = defaults_values.devicons
+   end
+ 
+   if config.values.devicons_enable then
+     config.values.devicons.enable = true
+     -- echo waring
+     vim.api.nvim_echo({ { '\n', 'Normal' } }, false, {})
+     vim.api.nvim_echo(
+       { { '[lir.nvim] `devicons_enable` is deprecated. Use `devicons.enable` instead. \n', 'WarningMsg' } },
+       false, {})
+   end
  end
  

```

File: lua/lir/config.lua [hunk 1]
```diff
@@ -28,13 +31,17 @@ local config = {}
 ---@class lir.config.values
 ---@field ignore            table
 ---@field show_hidden_files boolean
----@field devicons_enable   boolean
+---@field devicons          table
```
Comment (tamago324) [inline lua/lir/config.lua:34]: Please make the correction here.

```diff
- ---@field devicons          table
+ ---@field devicons          lir.config.values.devicons
```

**Review by tamago324 — CHANGES_REQUESTED** [review]

**Review by tamago324 — COMMENTED** [review]
I am sorry for repeating myself.
Please change `devicons.hishlight_dirnames` to `devicons.hishlight_dirname`.
The `float.curdir_window.highlight_dirname` doesn't have the `s` and I want to make it the same.

**Review by tamago324 — APPROVED** [review]

**Review by tamago324 — APPROVED** [review]
LGTM

**Comment by tamago324** [comment]
Thank you!

**Pull request merged**